### PR TITLE
moved ByteCodec and TaggedCodec to core

### PR DESCRIPTION
**Pull request by pshemass**

### CODE REVIEW

File: keeper/src/main/scala/zio/keeper/errors.scala [hunk 1]
```diff
@@ -15,16 +13,28 @@ sealed abstract class SerializationError(msg: String = "") extends Error(msg = m
 
 object SerializationError {
 
-  final case class SerializationTypeError[A](val cause: Throwable)(implicit ct: ClassTag[A])
+  final case class SerializationTypeError(msg0: String)
```
Comment (mschuwalow) [inline keeper/src/main/scala/zio/keeper/errors.scala]: we should probably also set the exception cause in order to preserve stack traces

File: keeper/src/main/scala/zio/keeper/membership/ByteCodec.scala [hunk 2]
```diff
@@ -17,9 +19,9 @@ object ByteCodec {
     new ByteCodec[A] {
 
       def toChunk(a: A) =
-        ZIO.effect(Chunk.fromArray(writeBinary(a)(rw))).mapError(SerializationError("Failed serializing to chunk.", _))
+        ZIO.effect(Chunk.fromArray(writeBinary(a)(rw))).mapError(SerializationTypeError(_))
```
Comment (mschuwalow) [inline keeper/src/main/scala/zio/keeper/membership/ByteCodec.scala]: I don't know whether calling this a TypeError is a good idea

Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/membership/ByteCodec.scala]: why?

Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/membership/ByteCodec.scala]: I will changed that

File: keeper/src/main/scala/zio/keeper/errors.scala (hunk 1 above)
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/errors.scala]: I will add that

**Pull request merged**